From 77882aa93f6df50f66eb2de92dfc0a160e718ece Mon Sep 17 00:00:00 2001
From: Weiguang Kong <weiguang.kong@nxp.com>
Date: Tue, 13 Mar 2018 18:40:06 +0800
Subject: [PATCH 3670/5242] MLK-17635-2: ASoC: fsl_dsp: fix unhandled
 alignment fault in user space

commit  d68008198f22c743ee635d9c9f638e693946bf09 from
https://source.codeaurora.org/external/imx/linux-imx.git

When using memcpy() or fread() function in dsp unit test or wrapper code,
an unhandled alignment fault error will occur randomly, this issue
is caused by the setting of mmap(). After using pgprot_writecombine()
function instead of pgprot_noncached() function, this error will not
occur.

Signed-off-by: Weiguang Kong <weiguang.kong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_hifi4.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/fsl/fsl_hifi4.c b/sound/soc/fsl/fsl_hifi4.c
index aa64536..b8a3abc 100644
--- a/sound/soc/fsl/fsl_hifi4.c
+++ b/sound/soc/fsl/fsl_hifi4.c
@@ -532,7 +532,7 @@ static int fsl_hifi4_mmap(struct file *file, struct vm_area_struct *vma)
 	size = hifi4_priv->scratch_buf_size;
 
 	/* ...remap shared memory to user-space */
-	vma->vm_page_prot = pgprot_noncached(vma->vm_page_prot);
+	vma->vm_page_prot = pgprot_writecombine(vma->vm_page_prot);
 	r = remap_pfn_range(vma, vma->vm_start, pfn, size, vma->vm_page_prot);
 	if (r != 0) {
 		pr_err("mapping failed: %d", r);
@@ -839,7 +839,7 @@ static int fsl_hifi4_probe(struct platform_device *pdev)
 	}
 
 	hifi4_priv->sdram_phys_addr = SDRAM_BASE_ADDR;
-	hifi4_priv->sdram_vir_addr = ioremap(hifi4_priv->sdram_phys_addr,
+	hifi4_priv->sdram_vir_addr = ioremap_wc(hifi4_priv->sdram_phys_addr,
 							SDRAM_BASE_SIZE);
 	if (!hifi4_priv->sdram_vir_addr) {
 		dev_err(&pdev->dev, "failed to remap sdram space for hifi4 firmware\n");
-- 
1.7.9.5

